cli/azd/pkg/tools/kubectl/util.go

```diff
@@ -0,0 +1,148 @@
+package kubectl
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	ErrResourceNotFound = errors.New("cannot find resource")
+	ErrResourceNotReady = errors.New("resource is not ready")
+)
+
+func GetResource[T any](
+	ctx context.Context,
+	cli KubectlCli,
+	resourceType ResourceType,
+	resourceName string,
+	flags *KubeCliFlags,
+) (T, error) {
+	if flags == nil {
+		flags = &KubeCliFlags{}
+	}
+
+	if flags.Output == "" {
+		flags.Output = "json"
+	}
+
+	var zero T
```

Comment:
nit,
giving the name `zero` doesn't mean we don't create an empty T.
Consider using `var result T` and using it for both `zero` and `resource`.  You can even name the return values like `(result T, err error)` and then you don't have to create an extra T